post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -257,37 +304,36 @@ public void shutdown() {
         processDatamartIds();
     }
 
-    private Long extractIdFromMessage(String topic, String messageKey) {
+    private String extractValFromMessage(String topic, String payload) {
         try {
-            logger.info("Got this key payload: {} from the topic: {}", messageKey, topic);
-            JsonNode keyNode = objectMapper.readTree(messageKey);
-
-            Entity entity = getEntityByTopic(topic);
-            if (Objects.isNull(keyNode.get(PAYLOAD).get(entity.getUidName()))) {
-                throw new NoSuchElementException("The '" + entity.getUidName() + "' value is missing in the '" + topic + "' message payload.");
-            }
-            return keyNode.get(PAYLOAD).get(entity.getUidName()).asLong();
-        } catch (Exception e) {
-            String msg = "Error processing '" + topic + "'  message: " + e.getMessage();
-            throw new RuntimeException(msg, e);
-        }
-    }
-
-    private String extractValFromMessage(String topic, String payload, String valName) {
-        try {
-            Entity entity = getEntityByTopic(topic);
-            if (entity.equals(Entity.INVESTIGATION)) {
-                JsonNode tblNode = objectMapper.readTree(payload).get(PAYLOAD).path(valName);
+            if (topic.endsWith(Entity.INVESTIGATION.getName())) {
+                JsonNode tblNode = objectMapper.readTree(payload).get(PAYLOAD).path("rdb_table_name_list");
                 if (!tblNode.isMissingNode() && !tblNode.isNull()) {
                     return tblNode.asText();
                 }
+            } else if (topic.endsWith(Entity.OBSERVATION.getName())) {
+                String domainCd = objectMapper.readTree(payload).get(PAYLOAD).path("obs_domain_cd_st_1").asText();
```

Comment:
Consider doing some NPE checks and log errors for any possible nulls.

Reply:
> Consider doing some NPE checks and log errors for any possible nulls.

No need in this case. Null `obs_domain_cd_st_1` and null `ctrl_cd_display_form` are pretty common, and it's not an error. Using `JsonNode path(String fieldName)` method is a good protection agains NPE, because it returns MIssingNode object for null or non-existing fieds. And MissingNode.asText() returns empty string.